src/classes/fflib_ApexMocksUtils.cls

```diff
@@ -91,6 +91,51 @@ public class fflib_ApexMocksUtils
         return deserializeParentsAndChildren(parentsType, parentDescribe, relationshipField, parents, children);
 	}
 
+    /**
+     * Gives the ability to set test values on formula
+     * and other read-only fields of mock SObjects
+     */
+    public static Object setReadOnlyField(SObject objectInstance, Type objectType, Schema.SObjectField field, Object value) {
```

Comment:
We have a similar method (local to our project) to accomplish this same task. It's more concise and doesn't require string concatenation on the JSON. Would this work in place of the two new methods you propose here?

```Java
	public SObject forceFieldWrite(SObject obj, Type deserializeType, Map<SObjectField, Object> properties) {
		Map<String, Object> mergedMap = new Map<String, Object>(obj.getPopulatedFieldsAsMap());
		for (SObjectField field : properties.keySet()) {
			// Merge the values from the properties map into the fields already set on the object
			mergedMap.put(field.getDescribe().getName(), properties.get(field));
		}
		// Serialize the merged map, and then deserialize it as the desired object type.
		String jsonString = JSON.serializePretty(mergedMap);
		return (SObject) JSON.deserialize(jsonString, deserializeType);
	}
```

Reply:
Presumably. Same 4 method parameters?

Reply:
Yes, I think that would work, and I like that it has one less parameter. You saying you're going to add it or you want me to?  If you plan on adding, will you also please include an example in the readme as I have done? Thanks!

Reply:
Can you try replacing your method with the one I propose to make sure it fits all of the use cases you are trying to address? 

I also don't entirely care about the name of the method; If you want to keep it as 'setReadOnlyField', you just need to make sure the name becomes plural since the method I'm proposing takes a collection of fields to assign.

Reply:
Done.